pkg/operator2/secret.go

```diff
@@ -17,13 +18,37 @@ func (c *osinOperator) expectedSessionSecret() *v1.Secret {
 	secret, err := c.secrets.Secrets(targetName).Get(targetName, metav1.GetOptions{})
 	if err != nil || isInvalidSessionSecret(secret) {
 		glog.V(4).Infof("failed to get secret %s: %v", targetName, err)
-		return randomSessionSecret()
+		generatedSessionSecret := randomSessionSecret()
+		if isInvalidSessionSecret(generatedSessionSecret) {
```

Comment:
Can we reverse this? (`if !isValidSessionSecret(...`). I almost missed that it was checking if it's *_In_valid*.

Reply:
yes, I also messed myself up with this in the test, for a few minutes I was all 'wait, why is this failing?? ...oooh, it's bc I'm testing if it's _in_ valid, not valid' : )